src/exec_auth.ts

```diff
@@ -99,15 +99,46 @@ export class ExecAuth implements Authenticator {
             exec.env.forEach((elt) => (env[elt.name] = elt.value));
             opts = { ...opts, env };
         }
-        const result = this.execFn(exec.command, exec.args, opts);
-        if (result.error) {
-            throw result.error;
-        }
-        if (result.status === 0) {
-            const obj = JSON.parse(result.stdout.toString('utf8')) as Credential;
-            this.tokenCache[user.name] = obj;
-            return obj;
-        }
-        throw new Error(result.stderr.toString('utf8'));
+
+        return new Promise((resolve, reject) => {
+            let stdoutData: string = '';
+            let stderrData: string = '';
+            let savedError: Error | undefined = undefined;
+
+            const subprocess = this.execFn(exec.command, exec.args, opts);
+            subprocess.stdout.setEncoding('utf8');
+            subprocess.stderr.setEncoding('utf8');
+
+            subprocess.stdout.on('data', (data: Buffer | string) => {
+                stdoutData += data.toString('utf8');
+            });
+
+            subprocess.stderr.on('data', (data) => {
+                stderrData += data.toString('utf8');
+            });
+
+            subprocess.on('error', (error) => {
+                savedError = error;
+                throw error;
```

Comment:
I'm not sure that it still makes sense to throw here.

Reply:
my mistake, I forgot to remove it